py3status/modules/file_status.py

```diff
@@ -68,16 +101,43 @@ def file_status(self):
                 'cached_until': self.py3.CACHE_FOREVER,
             }
 
-        if exists(self.path):
-            icon = self.icon_available
+        # init data
+        file_status_data = {}
+        # expand glob from paths
+        paths = list(map(glob, self.path))
+        # merge list of paths
+        paths = sorted([files for path in self.path for files in glob(path)])
+        file_status_data['paths'] = len(paths)
+
+        # fill data, legacy stuff
+        color = self.py3.COLOR_BAD
+        if paths:
             color = self.py3.COLOR_GOOD
-        else:
-            icon = self.icon_unavailable
-            color = self.py3.COLOR_BAD
+
+        # format paths
+        if self.init['format_path']:
+            self.format_path_separator = self.py3.safe_format(self.format_path_separator)
+
+            format_path = {}
+
+            if 'basename' in self.init['format_path']:
+                format_path['basename'] = map(basename, paths)
+
+            if 'fullpath' in self.init['format_path']:
+                format_path['fullpath'] = paths
+
+            for x in self.init['format_path']:
+                format_path[x] = self.py3.composite_join(
+                    self.format_path_separator, format_path[x]
+                )
+
+            file_status_data['format_path'] = self.py3.safe_format(
+                self.format_path, format_path
+            )
 
         response = {
             'cached_until': self.py3.time_in(self.cache_timeout),
-            'full_text': self.py3.safe_format(self.format, {'icon': icon}),
+            'full_text': self.py3.safe_format(self.format, file_status_data),
```

Comment:
(Opinion). May look cleaner if we pass `paths` and `format_path` here like the `icon`.